crates/ruff/src/linter.rs

```diff
@@ -526,3 +488,68 @@ This indicates a bug in `{}`. If you could open an issue at:
         });
     }
 }
+
+#[allow(clippy::print_stderr)]
+fn report_failed_to_converge_error(path: &Path, transformed: &str) {
+    #[cfg(debug_assertions)]
+    {
+        eprintln!(
+            "{}: Failed to converge after {} iterations in `{}`:---\n{}\n---",
+            "debug error".red().bold(),
+            MAX_ITERATIONS,
+            fs::relativize_path(path),
+            transformed,
+        );
+    }
+    #[cfg(not(debug_assertions))]
+    {
+        eprintln!(
+            r#"
+{}: Failed to converge after {} iterations.
+
+This indicates a bug in `{}`. If you could open an issue at:
+
+    {}/issues/new?title=%5BInfinite%20loop%5D
+
+...quoting the contents of `{}`, along with the `pyproject.toml` settings and executed command, we'd be very appreciative!
+"#,
+            "error".red().bold(),
+            MAX_ITERATIONS,
+            env!("CARGO_PKG_NAME"),
+            env!("CARGO_PKG_REPOSITORY"),
+            fs::relativize_path(path),
+        );
+    }
+}
```

Comment:
I recommend you to define the function once for `debug_assertions` and once for release to avoid the clippy errors:

```suggestion

#[cfg(debug_assertions)]
#[allow(clippy::print_stderr)]
fn report_failed_to_converge_error(path: &Path, transformed: &str) {
    eprintln!(
        "{}: Failed to converge after {} iterations in `{}`:---\n{}\n---",
        "debug error".red().bold(),
        MAX_ITERATIONS,
        fs::relativize_path(path),
        transformed,
    );
}

#[cfg(not(debug_assertions))]
fn report_failed_to_converge_error(path: &Path, _transformed: &str) {
  eprintln!(
      r#"
{}: Failed to converge after {} iterations.

This indicates a bug in `{}`. If you could open an issue at:

{}/issues/new?title=%5BInfinite%20loop%5D

...quoting the contents of `{}`, along with the `pyproject.toml` settings and executed command, we'd be very appreciative!
"#,
      "error".red().bold(),
      MAX_ITERATIONS,
      env!("CARGO_PKG_NAME"),
      env!("CARGO_PKG_REPOSITORY"),
      fs::relativize_path(path),
  );
}
```

Reply:
The clippy errors were for a previous version of this PR, CI seems green now.